addons/account_edi/models/account_edi_document.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields, api, _
+from psycopg2 import sql, OperationalError
+from contextlib import contextmanager
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountEdiDocument(models.Model):
+    _name = 'account.edi.document'
+    _description = 'Electronic Document for an account.move'
+
+    edi_format_id = fields.Many2one('account.edi.format')
+    edi_format_name = fields.Char(related='edi_format_id.name')
+    move_id = fields.Many2one('account.move')
+    state = fields.Selection([('to_send', 'To Send'), ('sent', 'Sent'), ('to_cancel', 'To Cancel'), ('cancelled', 'Cancelled')])
+    error = fields.Html()
+    attachment_id = fields.Many2one('ir.attachment')
+    ready_for_post_or_cancel = fields.Boolean(compute='_compute_ready_for_post_or_cancel', help='Technical field to now if this document is ready to be processed.')
```

Comment:
? I can find a single occurence of the usage of that field, and it would be easier to check if edi.document.state in ('to_send', 'to_cancel') or edi.document.state != move_id.state.

This needs a deeper investigation

Reply:
For an invoice it's enough to check if edi.document.state in ('to_send', 'to_cancel'), but for payments, we need additional checks like know if the related invoices' edi documents have already been sent.